src/Microsoft.DotNet.CodeAnalysis/Analyzers/PinvokeAnalyzer.cs

```diff
@@ -0,0 +1,157 @@
+        private const string Description = "The Pinvoke call should be removed.";
+        private const string AnalyzerName = "PinvokeAnalyzer";
+
+        private static DiagnosticDescriptor InvalidPinvokeCall = new DiagnosticDescriptor(DiagnosticIds.BCL0100.ToString(), Title, MessageFormat, AnalyzerName, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(InvalidPinvokeCall); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext obj)
+        {
+            AllowedPinvokeFile = obj.Options.AdditionalFiles.FirstOrDefault(f => Path.GetFileName(f.Path).Contains("PinvokeAnalyzer_"));
+            ExceptionFile = obj.Options.AdditionalFiles.FirstOrDefault(f => Path.GetFileName(f.Path).Contains("PinvokeAnalyzerExceptionList.analyzerdata"));
+            obj.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private Dictionary<string, string> _allowedPinvokes;
+        private Dictionary<string, string> _exceptionPinvokes;
+
+        private Dictionary<string, string> AllowedPinvokes
+        {
+            get
+            {
+                if (_allowedPinvokes == null)
+                {
+                    _allowedPinvokes = ParseAdditionalFile(AllowedPinvokeFile);
+                }
+
+                return _allowedPinvokes;
+            }
+        }
+
+        private Dictionary<string, string> ExceptionPinvokes
+        {
+            get
+            {
+                if (_exceptionPinvokes == null)
+                {
+                    _exceptionPinvokes = ParseAdditionalFile(ExceptionFile);
+                }
+
+                return _exceptionPinvokes;
+            }
+        }
+
+        private AdditionalText ExceptionFile { get; set; }
+        private AdditionalText AllowedPinvokeFile { get; set; }
+
+        private HashSet<string> _isNotSupportedOnWin7 = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private HashSet<string> IsNotSupportedOnWin7 { get { return _isNotSupportedOnWin7; } }
+
+
+        private Dictionary<string, string> ParseAdditionalFile(AdditionalText additionalFile)
+        {
+            Dictionary<string, string> parsedPinvokes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            if (additionalFile != null)
+            {
+                SourceText fileContents = additionalFile.GetText();
+                foreach (TextLine line in fileContents.Lines)
+                {
+                    string lineStr = line.ToString();
+                    if (!string.IsNullOrWhiteSpace(lineStr) && !lineStr.StartsWith("<!--"))
```

Comment:
The supported list format is pretty rudimentary. I have not handled cases where the comments would be multiline or present in the same line etc. However since I don't expect the files to be changed that often, I don't see any issue with it.